src/data/fintype/basic.lean

```diff
@@ -229,6 +229,91 @@ def of_surjective [decidable_eq β] [fintype α] (f : α → β) (H : function.s
   fintype β :=
 ⟨univ.image f, λ b, let ⟨a, e⟩ := H b in e ▸ mem_image_of_mem _ (mem_univ _)⟩
 
+end fintype
+
+section inv
+
+variables [fintype α] [decidable_eq β]
+
+section injective
+
+variables {f : α → β} (hf : function.injective f)
+
+/--
+The inverse of an `injective` function `f : α → β`, of the type `↥(set.range f) → α`.
+This is the computable version of `function.inv_fun` that requires `fintype α` and `decidable_eq β`.
+This function should not usually be used for actual computation because for most cases,
+an explicit inverse can be stated that has better computational properties.
+This function computes by checking all terms `a : α` to find the `f a = b`, so it is O(N) where
+`N = fintype.card α`.
+-/
+def function.injective.inv_of_mem_range : set.range f → α :=
```

Comment:
Does it make sense to just add a `namespace function` and `namespace injective` so that all the names become a bit shorter?